### PR TITLE
Use smaller base image, based on Alpine

### PR DESCRIPTION
**Pull request by pelithne**
PR related to https://github.com/Azure-Samples/azure-voting-app-redis/issues/32

The only change is that the base image is changed to a smaller, Alpine based, version. This make the whole app more light weight without changing any features.

The change has been manually tested to work with the current tutorial: https://docs.microsoft.com/en-us/azure/aks/tutorial-kubernetes-prepare-app

However, the `docker images` commands used in the tutorial will render a slightly different output

### CODE REVIEW

**Comment by Overv** [comment]
I don't see why this isn't being merged. It reduces the image size from 950 MB to 193 MB!

**Comment by pelithne** [comment]
I can make a pull request to the original repo. I think (not sure) that I made some other changes that should not be merged though 🙂

/peter

________________________________

**Comment by Overv** [comment]
@pelithne What do you mean by original repo?

**Comment by Overv** [comment]
@neilpeterson @iainfoulds What is stopping you from merging this?

**Comment by pelithne** [comment]
@Overv I must have commented in my sleep :-), please disregard!

**Comment by iainfoulds** [comment]
@Overv I wasn't a repo admin, so didn't receive notifications on this PR. I'm not sure why the original manifest used the larger Ubuntu image. We can prioritize this against other deliverables and review in order to merge it. Please be respectful in how you ask for assistance or updates.

**Comment by Overv** [comment]
@iainfoulds I asked this way because I just thought it was strange that such an simple and obvious improvement appeared to be overlooked. I was going through the tutorial for AKS last week and the first thing I wondered was why such a simple web app required me to push such a huge image.

File: .github/workflows/main.yml
Comment (zr-msft) [inline .github/workflows/main.yml:1]: Is this file needed to support the base image change? or is it out of scope of this PR?

File: azure-vote-all-in-one-redis.yaml
Comment (zr-msft) [inline azure-vote-all-in-one-redis.yaml:1]: many of these changes have been incorporated by other merged PRs. Please rebase.

File: azure-vote-all-in-one-redis.yaml [hunk 1]
```diff
@@ -42,9 +53,11 @@ spec:
       labels:
         app: azure-vote-front
     spec:
+      nodeSelector:
+        "beta.kubernetes.io/os": linux
       containers:
       - name: azure-vote-front
-        image: microsoft/azure-vote-front:v1
+        image: privatepelithne.azurecr.io/azuredocs/azure-vote-front:latest
```
Comment (zr-msft) [inline azure-vote-all-in-one-redis.yaml:60]: This should stay a reference to the image in MCR which will appear after a rebase:
https://github.com/Azure-Samples/azure-voting-app-redis/blob/master/azure-vote-all-in-one-redis.yaml#L60

File: azure-vote/azure-vote/config_file.cfg
Comment (zr-msft) [inline azure-vote/azure-vote/config_file.cfg:1]: Please leave the original values and revert this change.

File: workflow/main.yml
Comment (zr-msft) [inline workflow/main.yml:1]: similar to `.github/workflows`, is this needed to support the proposed image change? or is this out of scope?

**Review by zr-msft — CHANGES_REQUESTED** [review]
@pelithne thank you so much for this contribution and apologies for the delayed response.

Please take a look at my feedback/questions.

**Comment by phemmmie** [comment]
new update

**Pull request closed without merging**